src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.Analysis.cs

```diff
@@ -374,6 +289,62 @@ void RecordClosureScope(Scope innermost, Scope outermost, Closure closure)
             }
 
             /// <summary>
+            /// Walk up the scope tree looking for a closure.
+            /// </summary>
+            /// <returns>
+            /// A tuple of the found <see cref="Closure"/> and the <see cref="Scope"/> it was found in.
+            /// </returns>
+            public static (Closure, Scope) FindClosureInScope(Scope startingScope, MethodSymbol closureSymbol)
```

Comment:
Consider `GetClosureInScope` vs. `Find`. The general rule for funtions like this is:

- Find: returns null on failure. Essentially a method that expects failure as an outcome.
- Get: throws on failure. A method which expects to find the value.